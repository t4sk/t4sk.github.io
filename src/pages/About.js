import React from "react"

export function About(prop) {
  return <div>About</div>
}

export default About
