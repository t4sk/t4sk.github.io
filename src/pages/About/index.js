import React from "react"
import Layout from "../../components/Layout"
import SEO from "../../components/SEO"
import Html from "./index.html.js"

function Base(props) {
  const { lang } = props

  return (
    <Layout lang={lang}>
      <SEO title="About | Smart Contract Programmer" lang={lang} />
      <Html />
    </Layout>
  )
}

const About = {
  En: props => Base({ ...props, lang: "en" }),
  Jp: props => Base({ ...props, lang: "jp" }),
}

export default About
