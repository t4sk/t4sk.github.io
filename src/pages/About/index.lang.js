const LANG = {
  en: {
    About: "About",
    Contact: "Contact",
  },
  jp: {
    About: "概要",
    Contact: "お問い合わせ",
  },
}

export default LANG
