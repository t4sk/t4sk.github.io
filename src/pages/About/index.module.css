.Component {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.header {
  font-size: 2.5rem;
}
.links {
  display: flex;
  flex-direction: row;
}
.intro {
  color: #4e4e4e;
  font-size: 1.3rem;
  margin-top: 50px;
}
.email {
  font-weight: 700;
}
