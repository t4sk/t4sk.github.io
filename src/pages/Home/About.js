import React from "react"
import { Image, Icon, Divider } from "semantic-ui-react"
import styles from "./About.module.css"
import profile from "./profile.jpeg"

function About(props) {
  // TODO consulting and contact form

  return (
    <div className={styles.component}>
      <div className={styles.profile}>
        <Image
          src={profile}
          circular
          fluid
          style={{ width: 150, height: 150 }}
          alt="profile"
        />
        <h1 className={styles.header}>Tasuku Nakamura</h1>
      </div>
      <div className={styles.intro}>
        <p>
          Hi, I am a software engineer who makes videos about smart contracts.
        </p>
        <p>
          Creator of{" "}
          <a
            href="https://solidity-by-example.org/"
            target="__blank"
            rel="noopener noreferrer"
          >
            Solidity by Example
          </a>
        </p>
      </div>

      <Divider />

      <h3>Follow me</h3>
      <div>
        <a
          href="https://www.youtube.com/channel/UCJWh7F3AFyQ_x01VKzr9eyA/featured"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="youtube" color="red" size="large" />
            YouTube
          </div>
        </a>
        <a
          href="https://twitter.com/ProgrammerSmart"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="twitter" color="blue" size="large" />
            Twitter
          </div>
        </a>
        <a
          href="https://github.com/t4sk"
          target="__blank"
          rel="noopener noreferrer"
        >
          <div className={styles.followMeLink}>
            <Icon name="github" color="black" size="large" />
            GitHub
          </div>
        </a>
      </div>

      <Divider />
      <p>
        You can contact me at{" "}
        <span className={styles.email}>
          contact@smartcontractprogrammer.com
        </span>
      </p>
    </div>
  )
}

About.propTypes = {}

export default About
