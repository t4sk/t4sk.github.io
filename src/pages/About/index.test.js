import React from "react"
import { shallow } from "enzyme"
import About from "./index"

const props = {
  lang: "en",
}

test("en", () => {
  const component = shallow(<About.En {...props} />)

  expect(component).toMatchSnapshot()
})

test("jp", () => {
  const component = shallow(<About.Jp {...props} lang="jp" />)

  expect(component).toMatchSnapshot()
})
