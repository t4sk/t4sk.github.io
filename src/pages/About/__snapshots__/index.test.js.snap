// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`en 1`] = `
<Layout
  lang="en"
>
  <SEO
    description="Learn how to program smart contracts."
    keywords={
      Array [
        "smart contract",
        "solidity",
        "ethereum",
      ]
    }
    lang="en"
    meta={Array []}
    title="About | Smart Contract Programmer"
  />
  <_default />
</Layout>
`;

exports[`jp 1`] = `
<Layout
  lang="jp"
>
  <SEO
    description="Learn how to program smart contracts."
    keywords={
      Array [
        "smart contract",
        "solidity",
        "ethereum",
      ]
    }
    lang="en"
    meta={Array []}
    title="About | Smart Contract Programmer"
  />
  <_default />
</Layout>
`;
